src/main/java/com/google/api/codegen/config/ResourceNameMessageConfigs.java

```diff
@@ -28,20 +36,44 @@
 
   abstract ImmutableMap<String, ResourceNameMessageConfig> getResourceTypeConfigMap();
 
+  public abstract ListMultimap<String, Field> getFieldsWithResourceNamesByMessage();
```

Comment:
I can't tell what exactly the key is in this context - a String isn't === a Message.